### PR TITLE
Bug/add customer reference on install

### LINKED ISSUE #452: Add customer reference field when installing module

### PR DESCRIPTION
**Pull request by olsi-qose**
## Description
Add `customer reference` field when installing the module

## Related Issue
closes #452 

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have assigned the correct milestone or created one if non existent.
- [x] I have correctly labeled this pull request.
- [x] I have linked the corresponding issue in this description.
- [x] I have updated the corresponding Jira ticket.
- [ ] I have requested a review from at least 2 reviewers
- [x] I have checked the base branch of this pull request
- [x] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

File: Helper/Data.php [hunk 1]
```diff
@@ -656,4 +668,53 @@ public function clearMagentoCache($type)
             $this->cacheManager->clean($clearTypes);
         }
     }
+
+    /**
+     * @param ModuleDataSetupInterface $setup
+     * @throws LocalizedException
+     * @throws \Zend_Validate_Exception
+     */
+    public function addCustomerReference(ModuleDataSetupInterface $setup)
```
Comment (hannunosto) [inline Helper/Data.php]: Let's not put this into the data helper. I would create a similar Core class for data what we have for schema: https://github.com/Nosto/nosto-magento2/blob/develop/Setup/Core.php#L47. Then extend that in `UpgradeData` and `InstallData`

File: Setup/CoreData.php [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        $attribute->addData(
+            [
+                'attribute_set_id' => $attributeSetId,
+                'attribute_group_id' => $attributeGroupId,
+                'used_in_forms' => ['adminhtml_customer', 'customer_account_edit'],
+            ]
+        );
+
+        // @codingStandardsIgnoreLine
+        $attribute->save();
```
Comment (supercid) [inline Setup/CoreData.php]: @hannunosto @olsi-qose Do we wanna let it blow if this fails and stop the installation or should we wrap this in a try-catch?

File: Setup/InstallData.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2019 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Tagging\Setup;
+
+use Magento\Customer\Setup\CustomerSetupFactory;
```
Comment (supercid) [inline Setup/InstallData.php]: Seems to be unused

File: Setup/InstallData.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+use Magento\Eav\Model\Entity\Attribute\SetFactory as AttributeSetFactory;
```
Comment (supercid) [inline Setup/InstallData.php]: Seems to be unused

File: Setup/UpgradeData.php [hunk 5]
```diff
@@ -36,21 +36,19 @@
...
 use Magento\Framework\Setup\UpgradeDataInterface;
 use Nosto\Tagging\Helper\Account as NostoHelperAccount;
 use Nosto\Tagging\Helper\Url as NostoHelperUrl;
 use Magento\Framework\App\Config\Storage\WriterInterface;
 use Magento\Store\Model\ScopeInterface;
-use Magento\Customer\Model\Customer;
-use Magento\Customer\Setup\CustomerSetupFactory;
-use Magento\Framework\Setup\Patch\DataPatchInterface;
-use Magento\Eav\Model\Entity\Attribute\SetFactory as AttributeSetFactory;
 use Nosto\Tagging\Helper\Data as NostoHelperData;
```
Comment (supercid) [inline Setup/UpgradeData.php]: Seems to be unused since you removed on lines 146 and 163

File: Setup/CoreData.php (hunk 2 above)
Comment (hannunosto) [reply Setup/CoreData.php]: I think it can blow up. Otherwise it will cause issues runtime

Comment (supercid) [reply Setup/CoreData.php]: 👍

File: Setup/CoreData.php [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+        $attribute->addData(
+            [
+                'attribute_set_id' => $attributeSetId,
+                'attribute_group_id' => $attributeGroupId,
+                'used_in_forms' => ['adminhtml_customer', 'customer_account_edit'],
+            ]
+        );
+
+        try {
```
Comment (supercid) [inline Setup/CoreData.php]: Please revert this. See comment https://github.com/Nosto/nosto-magento2/pull/453#discussion_r286457337

**Pull request merged**